### ISSUE TITLE
Add support for Windows Server 2019

### ISSUE THREAD

**Issue by sixeyed** [Microsoft/mssql-docker#370]
The Windows Server Core Docker image is GA now at `mcr.microsoft.com/windows/servercore:1809` - it would be great to get SQL Server images for 1809 (we already have them for .NET Core via [PR 719](https://github.com/dotnet/dotnet-docker/pull/719)).

**Comment by vin-yu** [comment]
We're planning to have up-to-date/regularly updated SQL Server + Windows containers ( similar to our Linux images) that works with all editions in the future. At the moment, recommend building your own for now with something like this:

https://github.com/Microsoft/mssql-docker/blob/master/windows/mssql-server-windows-developer/dockerfile

**Comment by threenub** [comment]
Hi @vin-yu - Any word on when the Windows container support is moving forward? Why were standard and enterprise SKUs pulled from Windows containers in the first place? (im out of the loop on that).

**Comment by vin-yu** [comment]
Not yet - I'll update you and this thread when we have something ready to share. 
Standard and Enterprise container images (for Windows) were never released, and we didn't have those editions as we were still working on making them enterprise ready. 🔨

**Comment by threenub** [comment]
@vin-yu : any updates? Seems pretty crazy that MS arent prioritising native windows container support for SQL?

**Comment by ravindravv** [comment]
HI All,
I am using windows servers as docker pull mcr.microsoft.com/windows/servercore:ltsc2019 on my windows 2019 system. Can i know, what is the ms sql image and IIS image for the respective server core on windows 2019. 
 I am getting the error " The container operating system does not match the host operating system." while using the image microsoft/mssql-server-windows-developer.  Can i know what is the correct image and tag for windows/servercore:ltsc2019 on windows 2019. 
Thanks in Advance ... Ravindra

**Comment by ravindravv** [comment]
what is the SQL image and IIS image  for windows 2019? I am using windows servercore:lstc2019 image.
Thanks -- Ravindra

**Comment by PKrause79** [comment]
I need Windows 2019 support, too (1809)! Getting " The container operating system does not match the host operating system." error

**Comment by ChrML** [comment]
Also wish it was kept updated for Windows 1809. Currently I'm building from the github dockerfile, which works fine on 1809, but not refering to the dockerhub version directly is kind of a maintenance hassle.

**Comment by Pellic** [comment]
Any update or workaround to run sql server 2017 developer in win server 2019 (1089)?
@vin-yu could you provide a detailed explaination about the suggested link?

**Comment by ChrML** [comment]
@Pellic No news that I heard.

But I published a workaround built on the official Dockerfile. It won't be maintained, and will remove it once Microsoft publishes an official one.

https://hub.docker.com/r/chrml/mssql-server-windows-express

**Comment by Pellic** [comment]
Thanks ChrML but I need the developer edition. There is a way to adapt the microsoft/mssql-server-windows-developer image?
Anyway when I run the pull from the suggested link I get the following error:
 docker : Error response from daemon: manifest for chrml/mssql-server-windows-express:latest not found
At line:1 char:1

**Comment by ChrML** [comment]
@Pellic Look at Tags. You need the :1809 tag, not :latest.

You can build a 1809 image for Developer using the Dockerfile Microsoft has published on Github. Just replace the base image in the Dockerfile with 1809. Or I can do it later this week.

**Comment by vin-yu** [comment]
Hi Everyone, I am running the SQL Server on Windows Containers (1809) in private preview very soon. 
We have a sign up form now at **http://aka.ms/sqleap** 

Please let me know what you're planning to do with SQL Server in Windows Containers under
 "Please describe the specific application or workload that you will be testing with SQL Server 2019?"

and select **SQL Server in Windows Containers** under
"Are there any other specific features you will test and validate?" 

@Pellic

**Comment by Pellic** [comment]
Thank you ChrML and vin-yu. I solved as suggested, building the image with some minor changes in the dockerfile

**Comment by 93Kamuran** [comment]
@pellic can you please share your solution with me? I have to use developer edition in windows server 2019 too.

**Comment by richardgavel** [comment]
I have a need for an older version of SQL Server (2016 SP1), but on the the windowsservercore:1903 image. I mounted the install files into the container and changed the base image. However, SQL Server itself doesn't seem to be installing (this has worked before for windowsservercore:1803). Unfortunately, the log file is so big, I'm having trouble figuring out where the problem is.

**Comment by kkbruce** [comment]
Hi guys

I published v1809, v1903 built on the official Dockerfile.  

https://hub.docker.com/r/kkbruce/mssql-server-windows-express

![v1809](https://user-images.githubusercontent.com/2024682/63988410-b816e680-cb0e-11e9-8ad9-19da6125ae7e.png)

![v1903](https://user-images.githubusercontent.com/2024682/63988434-de3c8680-cb0e-11e9-92ab-8837c387250e.png)

**Comment by codymullins** [comment]
For those having an issue on the mismatch of the container host, I was able to get around this by changing the FROM line to:

```
FROM mcr.microsoft.com/windows/servercore:ltsc2019
```

I'm assuming this will work if you change it to whatever version you need.

**Comment by KGitanjali** [comment]
Hi Pellic,

can you please share your docker file?

**Comment by Carpinteiro** [comment]
Yes , it would be very nice to see the adopted Dockerfile. I am trying to do it by myself but the execution of "SQL.exe" does nothing. No output , no error and does not install nothing

**Comment by theramis** [comment]
> Yes , it would be very nice to see the adopted Dockerfile. I am trying to do it by myself but the execution of "SQL.exe" does nothing. No output , no error and does not install nothing

@Carpinteiro you have probably fallen victim to this https://support.microsoft.com/en-us/help/4542617/you-might-encounter-issues-when-using-windows-server-containers-with-t. Specifically symptom number 4.

You can use a newer version of the docker image. Something like `FROM mcr.microsoft.com/windows/servercore:10.0.18363.720` which should fix the issue of `SQL.exe` doing nothing.

**Comment by KGitanjali** [comment]
@Carpinteiro ,
refer the link  **https://github.com/microsoft/mssql-docker/tree/master/windows/mssql-server-windows-express**  for docker file.
Use below base image instead of microsoft/windowsservercore in the docker file given on above link.
mcr.microsoft.com/windows/servercore:ltsc2019

I have tried this and it is working as expected.SQL expression edition get installed in windows container on windows 2019 server.

If you want to keep container in running state , add below line at the end of start.ps1
C:\ServiceMonitor.exe w3svc

**Comment by Carpinteiro** [comment]
Thank you @KGitanjali and @theramis for your answers. 

We were able to solve it by matching exactly the version of windows server with the version of our windows server machine.

My mistake was that I tried to use the version 1809 that appeared in the list of the image below from [Windows Server Core](https://hub.docker.com/_/microsoft-windows-servercore) to match the version displayed in system settings: 
![image](https://user-images.githubusercontent.com/8053153/80542499-fc5fcc80-89a4-11ea-979c-a480bafbe52d.png)



**BUT** the version of my windows server is 10.0.17763.973. Found it using **_ver_** command from a command line. 

I was not aware this image version was available because i needed to click on the [link highlighted in the image](https://mcrflowprodcentralus.cdn.mscr.io/mcrprod/windows/servercore?P1=1588111963&P2=1&P3=1&P4=WN7RzBwTdrcIklly%2BvPAE8bcKNMJemxl2I7ZBkmGodw%3D&se=2020-04-28T22%3A12%3A43Z&sig=1ixb7KW336n%2Fv9CXyaN7GZp56I1gDHw3xZieBZFE7vw%3D&sp=r&sr=b&sv=2015-02-21) to see all the versions available.

![image](https://user-images.githubusercontent.com/8053153/80543396-b1df4f80-89a6-11ea-9f45-d91c831abe43.png)


Thanks a lot!

**Comment by bestickley** [comment]
Why isn't SQL Server 2019 available on windows containers? I'd expect to see an updated 2019 version here:
https://hub.docker.com/r/microsoft/mssql-server-windows-developer/
Any one know?
